### ISSUE TITLE
Non-standard syntax compatibility

### ISSUE THREAD

**Issue by mturco** [ismay/stylelint-no-unsupported-browser-features#36]
Has this plugin been updated for Stylelint v8 compatibility? The README states v5+ support but I'm getting the following error on any LESS file that has an `@import` in it:

```
PS C:\Users\<user>\Desktop\stylelint-test-case> yarn run lint
yarn run v1.0.1
$ stylelint src/**/*.less
Error: Unkonwn node type import
    at C:\Users\<user>\Desktop\stylelint-test-case\node_modules\doiuse\lib\detect-feature-use.js:152:19
    at Root.each (C:\Users\<user>\Desktop\stylelint-test-case\node_modules\postcss-less\node_modules\postcss\lib\container.js:114:22)
    at Detector.node (C:\Users\<user>\Desktop\stylelint-test-case\node_modules\doiuse\lib\detect-feature-use.js:137:13)
    at Detector.process (C:\Users\<user>\Desktop\stylelint-test-case\node_modules\doiuse\lib\detect-feature-use.js:163:12)
    at Object.postcss (C:\Users\<user>\Desktop\stylelint-test-case\node_modules\doiuse\lib\doiuse.js:40:23)
    at postCssPlugin (C:\Users\<user>\Desktop\stylelint-test-case\node_modules\stylelint-no-unsupported-browser-features\index.js:35:27)
    at Promise.resolve.then (C:\Users\<user>\Desktop\stylelint-test-case\node_modules\stylelint\lib\lintSource.js:161:9)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Here's a [reduced test case](https://github.com/mturco/stylelint-test-case).

Downgrading to `stylelint@^7.0.0` fixes this issue.

**Comment by ismay** [comment]
I'm using it with stylelint v8 without issues. Don't have time for this now, but you could verify by cloning this repo and running npm test after manually installing stylelint v8.

**Comment by ismay** [comment]
Yep. It's tested with stylelint 8, so basic compatibility should be fine. Not sure about the way stylelint handles non standard syntax though, maybe that's an edgecase I've missed.

**Comment by ismay** [comment]
However if it works with stylelint 7 I'm not sure if it depends on the fact that you're using LESS. So that's weird.

**Comment by mturco** [comment]
Just tried a few things after reading your comments. The first was making the `@import` a standard CSS import for a web font:

```less
@import url('https://fonts.googleapis.com/css?family=Roboto');

.foo {
  display: flex;
}
```

That resulted in the same error. The second was changing the extension of the file to `.css` and that fixed it:
```
PS C:\Users\<user>\Desktop\stylelint-test-case> yarn run lint
yarn run v1.0.1
$ stylelint src/**/*.css

src/main.css
 4:3  ⚠  Unexpected browser feature "flexbox" is only partially supported by IE 10,11, Blackberry     plugin/no-unsupported-browser-features
         Browser 7, IE Mobile 10, UC Browser for Android 11.4

Done in 2.64s.
```

Same goes for changing the extention to `.scss`. So it does seem to be specific to LESS for whatever reason.

**Comment by ismay** [comment]
Hmm ok. So different behaviour based on the extension. My assumption is that stylelint is using something like a processor to parse the LESS. Maybe you've uncovered a bug in that.

Or maybe it's something I'm doing in this lib.

**Comment by ismay** [comment]
Ok so it uses postcss-less for parsing: https://github.com/stylelint/stylelint/blob/master/docs/user-guide/css-processors.md

**Comment by MatissJanis** [comment]
I'm running in to the same issue. Any fix?

**Comment by ismay** [comment]
@MatissJanis Apparently stylelint has utilities that check whether the syntax being checked is standard syntax (or non-standard like less, scss, sugar, etc.). Implementing that should fix this bug.

https://stylelint.io/developer-guide/rules/#write-the-rule

**Comment by MatissJanis** [comment]
Yeah, I tried that and it did not solve the issue.

**Comment by ismay** [comment]
> Yeah, I tried that and it did not solve the issue.

I mean implementing in this plugin though. Not in your own project. Did you test by modifying the source for this plugin?

**Comment by MatissJanis** [comment]
Ah.. no, sorry, I did not test it by modifying this plugin. Tried adding source type - `less`.

**Comment by ismay** [comment]
Should be fixable by removing [this error in doiuse](https://github.com/anandthakker/doiuse/blob/master/src/detect-feature-use.js#L132). I'll open a pr there, and if it passes the review this should be fixed by upgrading doiuse.

Error only occurs with less because the postcss less parser introduced a new node type that doiuse doesn't recognise. Which then causes it to throw an error.

**Comment by ismay** [comment]
Released as 1.0.1. I'd love to hear if this fixed it @mturco!

**Comment by mturco** [comment]
@ismay I can verify this is no longer an issue in 1.0.1. Thanks for fixing it!
